### DISCUSSION TITLE
Possibility to add torus connector

### DISCUSSION THREAD

**Discussion by OlremProj** [rainbow-me/rainbowkit#459] [Q&A]
Hi, is it possible to send PR to add torus connector ? We would like to use your library because it perfectly fits to our needs, but torus connector is missing. Can I add it on a fork and send you my PR ?

**Comment by nickbytes** [comment]
@OlremProj I'm not familiar w/ implementation specifics for Torus, but would love to see you + community explore making this work. We would welcome a PR! Reach out if you have any questions as you work through it!

**Reply by OlremProj** [reply]
@nickbytes Thanks for you quick answer!
The main differences are the followings:

- it doesn’t allow to switch chain
- its connector sends back a promise
- it opens a modal to connect

The main benefit of Torus is that there is no need for the user to download a browser extension. It also permits to create a wallet in seconds with an Oauth authentication (like Google, Facebook…)
I’ll make it work but before making all the tests and the refactor, I would like to send a PR for you to check if I’m in the good direction.
Is that ok for you?

**Reply by nickbytes** [reply]
yep, works for us!

**Reply by OlremProj** [reply]
[PR](https://github.com/rainbow-me/rainbowkit/pull/474)

**Reply by OlremProj** [reply]
@nickbytes hi, i would tel you I deploy the fork on npm to be able to use easily your library with ma torus connector during the PR is pending. 

https://www.npmjs.com/package/rainbow-fork
